app/assets/javascripts/Patient.js

```diff
@@ -0,0 +1,143 @@
+var Schema = mongoose.Schema;
+var Code = Schema.Types.Code;
+var Interval = Schema.Types.Interval;
+var Quantity = Schema.Types.Quantity;
+
+var PatientSchema = new Schema({
+  birth_datetime: DateTime,
+  qdm_version: { type: String, default: "5.3" },
+  ethnicity: Code,
+  race: Code,
+  sex: Code,
+  // These are the "data criteria", or QDM datatype elements that exist on a
+  // patient.
+  history_elements: []
+});
+
+// Returns an array of history elements that exist on this patient, that
+// match the given HQMF data criteria OID.
+PatientSchema.methods.get_by_hqmf_oid = function get_by_hqmf_oid(hqmf_oid, callback) {
+  return this.history_elements.filter(element => element.hqmf_oid === hqmf_oid);
+}
+
+// Returns an array of history elements that exist on this patient. Optionally
+// takes a category, which returns all history elements of that QDM category.
+// Example: patient.get_history_elements(category = 'encounters') will return
+// all Encounter QDM data types active on the patient.
+PatientSchema.methods.get_history_elements = function get_history_elements(category, callback) {
+  return this.history_elements.filter(element => element.category === category);
+}
+
+PatientSchema.methods.adverse_events = function adverse_events(params, callback) {
+  return this.get_history_elements('adverse_event');
+}
+
+PatientSchema.methods.assessments = function assessments(params, callback) {
```

Comment:
this method seems a bit like the HDS way which has gotten us into trouble. I think it could be different here because it directly maps to the QDM model and isn't related to parsing things, but it makes me nervous based on the issues we've had.

There should be a way to access by OID. If that were present, then both the oid method and this would be fine.

Reply:
These methods are just helper methods to grab data criteria on the patient by QDM category (Cypress needs this). This is different than the way things are with HDS at the moment, where the data_criteria are stored in collections dictated by their types (rather than in a single collection, like it is with this approach). This would make things easier for Bonnie (we wouldn't have to do things like this anymore: https://github.com/projecttacoma/cql_qdm_patientapi/blob/master/app/assets/javascripts/cqlpatient.js.coffee#L156).

There is a method above to grab by HQMF OID, is that what you mean?

Let me know if I misunderstood your question.

Reply:
totally missed the hqmf oid query possibility! thanks for pointing that out!

Reply:
These are just helper methods to grab the criteria out of `history_elements` by QDM category (Cypress needs this). This differs from the way HDS does it currently (criteria are stored on the record by category: https://github.com/projectcypress/health-data-standards/blob/master/lib/health-data-standards/models/record.rb#L31). With this approach we are putting them all in one, but still allowing you to look them up by category. This will make things easier for Bonnie, as we won't need to do this anymore: https://github.com/projecttacoma/cql_qdm_patientapi/blob/master/app/assets/javascripts/cqlpatient.js.coffee#L156.

There is a `find_by_hqmf_oid` method above, is that what you mean?

Let me know if I misunderstood your question.

Reply:
this is like deja vu... yes, i see the find_by_hqmf_oid now :) saw it in the previous post although i guess my post didn't make that clear.

Reply:
Ahh! Sorry, Github for some reason was not showing my original response (or your reply). I thought I forgot to submit the original post (and proceeded to re-write it)!

Reply:
ha! :)